flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamContextEnvironment.java

```diff
@@ -81,13 +60,15 @@ public JobExecutionResult execute(String jobName) throws Exception {
 
 		transformations.clear();
 
-		// execute the programs
-		if (wait) {
-			return client.runBlocking(streamGraph, jars, classpaths, userCodeClassLoader);
+		if (ctx.isWait()) {
+			return ctx.getClient().runBlocking(streamGraph, ctx.getJars(), ctx.getClasspaths(),
+					ctx.getUserCodeClassLoader());
 		} else {
-			JobSubmissionResult result = client.runDetached(streamGraph, jars, classpaths, userCodeClassLoader);
+			ctx.setCurrentPlan(streamGraph);
```

Comment:
Here, the last call to execute() would override all the previous ones. I think you should throw an Exception here like in ContextEnvironment. Or you throw the Exception on setCurrentPlan. We want to allow only one job to be submitted detached, right?